### PR TITLE
Mo todo

### PR DESCRIPTION
**Pull request by MohamedLebda**
added class Controls

### CODE REVIEW

**Comment by MohamedLebda** [comment]
added task class

File: javascript/todo/src/controls.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export default class Controls {
+  #titles;
```
Comment (AhmedLebda) [inline javascript/todo/src/controls.js]: No need for private properties, they aren't inherited so we won't have direct access for title and description properties

File: javascript/todo/src/controls.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  #dis;
+  constructor(title, discription) {
```
Comment (AhmedLebda) [inline javascript/todo/src/controls.js]: Fix typos : 
titles => title
discription => description

recommended: 
code spell checker extension for vscode 
Extension ID : streetsidesoftware.code-spell-checker

File: javascript/todo/src/controls.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    this.#titles = title;
+    this.#dis = discription;
```
Comment (AhmedLebda) [inline javascript/todo/src/controls.js]: Try to make your variable names more understandable
recommended => this.description = description;

File: javascript/todo/src/controls.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  getTitle() {
+    return console.log(this.#titles);
```
Comment (AhmedLebda) [inline javascript/todo/src/controls.js]: You can just remove 'return'.

File: javascript/todo/src/controls.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  }
+  setTitle(title) {
+    this.#titles = title;
+  }
+  getDis() {
+    return console.log(this.#dis);
```
Comment (AhmedLebda) [inline javascript/todo/src/controls.js]: You can just remove 'return'

File: javascript/todo/src/controls.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  }
+  setDis(discription) {
```
Comment (AhmedLebda) [inline javascript/todo/src/controls.js]: Try to write more understandable method names:
recommended: 
setDis(discription) => setDescription(newDescription)

**Review by AhmedLebda — CHANGES_REQUESTED** [review]
Summary : 
- More understandable and clear property and method names
- No need for private class fields
-  Fix typos

File: javascript/todo/src/task.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import Controls from "./controls";
+
+class Task extends Controls {
```
Comment (AhmedLebda) [inline javascript/todo/src/task.js:3]: class Task need to be exported.

File: javascript/todo/src/task.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  #date;
+  #order;
+  constructor(title, discription, dueDate, priority) {
```
Comment (AhmedLebda) [inline javascript/todo/src/task.js]: - Default values for the properties
- fix typos

File: javascript/todo/src/task.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import Controls from "./controls";
+
+class Task extends Controls {
+  #date;
```
Comment (AhmedLebda) [inline javascript/todo/src/task.js]: No need for private class fields

File: javascript/todo/src/task.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  #order;
+  constructor(title, discription, dueDate, priority) {
+    super(title, discription);
+    this.#date = dueDate;
+    this.#order = priority;
+  }
+  getDueDate() {
```
Comment (AhmedLebda) [inline javascript/todo/src/task.js:10]: You can remove 'return'

File: javascript/todo/src/task.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    return console.log(this.#date);
+  }
+  setDueDate(dueDate) {
+    this.#date = dueDate;
+  }
+  getPriority() {
+    return console.log(this.#order);
```
Comment (AhmedLebda) [inline javascript/todo/src/task.js]: you can remove 'return'

File: javascript/todo/src/task.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  }
+  setPriority(pirority) {
```
Comment (AhmedLebda) [inline javascript/todo/src/task.js]: setPriority method needs more validation to make sure users enter a number 
recommended : 
number should be between 1 - 10

**Review by AhmedLebda — CHANGES_REQUESTED** [review]
Summary : 
- export class Task
- setPriority method validation
- fix typos

**Pull request closed without merging**